pep-9999.rst

```diff
@@ -180,6 +180,17 @@ Yes::
         def do_stuff(self, y: str) -> None: ...
         doStuff = do_stuff
 
+The type of generic classes can be narrowed by annotating the ``self``
```

Comment:
If I'm understanding this PEP correctly, it's specific to type stubs. By adding this section to this PEP, should I assume that this usage of a "self" annotation within an overload should be honored only within a type stub? I would think it should apply within ".py" files as well. If that's true, then this probably belongs in an addendum or edit to PEP 484.

Reply:
I admit that that's a bit of a cop-out on my part. I don't know if this can still be added to PEP 484. On the other hand, I am not sure whether a full PEP is warranted for this change.

Reply:
Do any type checkers currently implement this?

Reply:
mypy and pyright do. I think pyre does too, but I'm not sure about that.